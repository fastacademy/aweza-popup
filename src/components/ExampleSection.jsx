import { h, Component } from 'preact'
import { createAudioBlobUrl } from '../util'
import Speaker from './Speaker'

export default class ExampleSection extends Component {
  render () {
    return (
      <div class="aweza-example">
        <div class="aweza-section-title">
          <b>{this.props.term.language.example_label}</b>
        </div>
        <div class="aweza-section-content">
          <ExampleText {...this.props.term}/>
          <hr/>
          <ExampleText {...this.props.translationTerm}/>
        </div>
      </div>
    )
  }
}

class ExampleText extends Component {
  render () {
    if (!this.props.example) {
      return
    }
    let audioUrl = null

    if (this.props.example_audio) {
      audioUrl = this.props.example_audio
    }
    else if (this.props.tts.example) {
      audioUrl = createAudioBlobUrl(this.props.tts.example)
    }

    return (
      <div class="aweza-example-text aweza-section-content-text">
        <Speaker uniqueId={`aweza-term-${this.props.id}-example-audio`} audioUrl={audioUrl}/>
        <div class="aweza-example-content-text-text">
          <div dangerouslySetInnerHTML={{__html: this.props.example}}/>
        </div>
        <div>
          <div class="aweza-example-content-text-right language-code">
            {this.props.language.code.toUpperCase()}
          </div>
        </div>
      </div>
    )
  }
}
