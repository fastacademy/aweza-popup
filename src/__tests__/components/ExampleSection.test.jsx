// noinspection ES6UnusedImports
import {h} from 'preact';
import render from 'preact-render-to-json';
import {terms} from '../dummydata';
import ExampleSection from "../../components/ExampleSection";

function getComponent() {
    return (
        <ExampleSection
            term={terms[0]}
            translationTerm={terms[0].translations[0]}
        />
    );
}

test('ExampleSection component', () => {
    global.URL.createObjectURL = jest.fn();

    const tree = render(getComponent());

    expect(tree).toMatchSnapshot();
});
