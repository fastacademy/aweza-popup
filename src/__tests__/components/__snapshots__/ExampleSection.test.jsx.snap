// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleSection component 1`] = `
<div
  class="aweza-example"
>
  <div
    class="aweza-section-title"
  >
    <b>
      Example
    </b>
  </div>
  <div
    class="aweza-section-content"
  >
    <hr />
    <div
      class="aweza-example-text aweza-section-content-text"
    >
      <div
        class="aweza-example-content-text-text"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "This is an example for parabola",
            }
          }
        />
      </div>
      <div>
        <div
          class="aweza-example-content-text-right language-code"
        >
          EN
        </div>
      </div>
    </div>
  </div>
</div>
`;
